### ISSUE TITLE
Use the m4v extension for Dash video files

### ISSUE THREAD

**Issue by Cycor** [rg3/youtube-dl#1808]
When downloading files I usually add a bunch of types, so i'm sure i'm at least downloading something. 

youtube-dl --no-overwrites --restrict-filenames --ignore-errors -o "./%(uploader_id)s/%(upload_date)s %(uploader)s - %(title)s.%(ext)s" -f 137/136/135/134/133/38/37/22/18/17 
youtube-dl --no-overwrites --restrict-filenames --ignore-errors -o "./%(uploader_id)s/%(upload_date)s %(uploader)s - %(title)s.%(ext)s" -f 141/140/139 

The problem i'm running into is that I have to join the mp4/m4a Dash files to get a complete video, now in most cases this will produce a video only mp4 and a audio m4a.

In some cases this produces a lower quality "normal" mp4 file ( when Dash isn't available ), now i have no easy way to recognize Dash files.

so a simple change would be to change the extension of the Dash video files to mp4v

ps. My original suggestion would be m4v since we already got m4a but it seems Apple uses this format.

**Comment by montanaflynn** [comment]
We need some way to tell youtube-dl to not use dash videos, I don't know of anyway currently but having the ext selector would do the job.

`$ youtube-dl -f '18/best[ext=mp4]'`

**Comment by Saklad5** [comment]
macOS treats `m4v` files differently than `mp4` files for some reason. It will parse more metadata, such as the artist field, if the extension is `.m4v`. This is extremely important when searching for files, so I’ve changed my personal script to rename `mp4` files to `m4v` after downloading them with `youtube-dl`.

For obvious reasons, I think it would be better to have the option in `youtube-dl` to use `m4v` from the start. Since the extension is identical to `mp4`, it should be trivial to change.

I’ve tried changing it myself, but I’m clearly missing something about the way the post-processing is coded. It should be easy for anyone that understands that though, from what I can tell.

**Comment by svnpenn** [comment]
@Saklad5 cant you just use:

~~~sh
--output '%(title)s.m4v'
~~~

or similar?

**Comment by Saklad5** [comment]
I’ve tried. Oddly enough, that ends up producing a file ending with `.mp4.m4v`. Not exactly what I’m looking for.

**Comment by svnpenn** [comment]
Hm ok then, that is pretty fucking stupid. The user should have control over the full filename, including the extension

**Comment by Saklad5** [comment]
@svnpenn Yeah, that clearly isn’t intended behavior. If it was, we wouldn’t have the `%(ext)s.` placeholder. Again, I suspect the post-processing code is to blame.

**Comment by svnpenn** [comment]
@Saklad5 you might want to check your version. I just tested this and it is not working as you say:

~~~
$ youtube-dl -f 140 --output example.m4v JkpT-xiSTBE
[youtube] JkpT-xiSTBE: Downloading webpage
[youtube] JkpT-xiSTBE: Downloading video info webpage
[youtube] JkpT-xiSTBE: Extracting video information
[download] Destination: example.m4v
[download] 100% of 5.25MiB in 00:00
[ffmpeg] Correcting container in "example.m4v"
~~~

**Comment by Saklad5** [comment]
Sorry, I may have forgotten to mention that I’m using the `--recode-video mp4` argument. That’s the post-processing I’m talking about.

**Comment by svnpenn** [comment]
@Saklad5 sorry, I am done here. I dont deal with moving goalpost. You on your own.

**Comment by Saklad5** [comment]
@svnpenn Fair enough. Sorry for the confusion.

**Comment by 89z** [comment]
Issue should be closed, no example given
